cg/services/deliver_files/file_formatter/utils/mutant_sample_service.py

```diff
@@ -0,0 +1,73 @@
+from pathlib import Path
+
+from cg.apps.lims import LimsAPI
+from cg.services.deliver_files.file_fetcher.models import SampleFile
+from cg.services.deliver_files.file_formatter.models import FormattedFile
+from cg.services.deliver_files.file_formatter.utils.sample_concatenation_service import (
+    SampleFileConcatenationFormatter,
+)
+from cg.services.fastq_concatenation_service.fastq_concatenation_service import (
+    FastqConcatenationService,
+)
+
+
+class MutantFileFormatter(SampleFileConcatenationFormatter):
+    def __init__(self, lims_api: LimsAPI, concatenation_service: FastqConcatenationService):
+        self.lims_api: LimsAPI = lims_api
+        super().__init__(concatenation_service=concatenation_service)
+
+    def format_files(
+        self, moved_files: list[SampleFile], ticket_dir_path: Path
+    ) -> list[FormattedFile]:
+        formatted_files: list[FormattedFile] = super().format_files(
+            moved_files=moved_files, ticket_dir_path=ticket_dir_path
+        )
+        formatted_files = self._add_lims_metadata(
+            formatted_files=formatted_files, sample_files=moved_files
+        )
+        unique_formatted_files = self._filter_unique_path_combinations(formatted_files)
+        return self._format_sample_files(unique_formatted_files)
+
+    def _get_lims_naming_metadata(self, sample_id: str) -> str:
+        region_code = self.lims_api.get_sample_attribute(
+            lims_id=sample_id, key="region_code"
+        ).split(" ")[0]
+        lab_code = self.lims_api.get_sample_attribute(lims_id=sample_id, key="lab_code").split(" ")[
+            0
+        ]
+        return f"{region_code}_{lab_code}_"
+
+    def _add_lims_metadata(
+        self, formatted_files: list[FormattedFile], sample_files: list[SampleFile]
+    ) -> list[FormattedFile]:
+        for formatted_file in formatted_files:
+            sample_id: str = self._get_sample_id_by_original_path(
+                original_path=formatted_file.original_path, sample_files=sample_files
+            )
+            lims_meta_data = self._get_lims_naming_metadata(sample_id)
+            formatted_file.original_path = formatted_file.formatted_path
```

Comment:
This is very sneaky to me. So after calling `_add_lims_metadata`, `[file.original_path for file in formatted_files]` will be `[file.formatted_path for file in formatted_files]` for whatever is in the input?

Reply:
Maybe this ties together with what @Vince-janv mentioned when talking about BaseModels - one of the core concepts of the base model is that it is supposed to be static and not changing, and I do think that it is quite hard to follow the steps that the SampleFile and FormattedFile objects undergo in this service.

Reply:
I will re work this flow. It is indeed unclear and that has to do with the inheritance in this case.

About the basemodels I agree that they should be static, but I think that refers to their structure (i.e. the FIELDS, themselves) and not necessarily the data within those fields. As long as they are valid for what can be place on those fields.

That being said. I do think this flow suffers from re-using the same base model with the same static structure and use it in a slightly different context. I will try to rectify that.